hs_discover/views.py

```diff
@@ -135,41 +138,37 @@ def get(self, request, *args, **kwargs):
+            authors = sqs.facet('author').facet_counts()['fields']['author']
+            owners = sqs.facet('owner').facet_counts()['fields']['owner']
+            subjects = sqs.facet('subject').facet_counts()['fields']['subject']
+            contributors = sqs.facet('contributor').facet_counts()['fields']['contributor']
+            types = sqs.facet('resource_type').facet_counts()['fields']['resource_type']
+            availability = sqs.facet('availability').facet_counts()['fields']['availability']
+            if request.GET.get('updatefilters'):
+                authors = [x for x in authors if x[1] > 0]
+                owners = [x for x in owners if x[1] > 0]
+                subjects = [x for x in subjects if x[1] > 0]
+                contributors = [x for x in contributors if x[1] > 0]
+                types = [x for x in types if x[1] > 0]
+                availability = [x for x in availability if x[1] > 0]
+            filterdata = [authors[:self.filterlimit], owners[:self.filterlimit], subjects[:self.filterlimit],
+                          contributors[:self.filterlimit], types[:self.filterlimit], availability[:self.filterlimit]]
+
         if sort == 'author':
             sqs = sqs.order_by('author_exact')
         elif sort == '-author':
             sqs = sqs.order_by('-author_exact')
         else:
             sqs = sqs.order_by(sort)
 
-        # Return counts for filter integers beside checkboxes
-        sqs_author = sqs.facet('author')
-        authors = sqs_author.facet_counts()['fields']['author'][:self.filterlimit]
-        authors = [x for x in authors if x[1] > 0]
-        sqs_owner = sqs.facet('owner')
-        owners = sqs_owner.facet_counts()['fields']['owner'][:self.filterlimit]
-        owners = [x for x in owners if x[1] > 0]
-        sqs_subject = sqs.facet('subject')
-        subjects = sqs_subject.facet_counts()['fields']['subject'][:self.filterlimit]
-        subjects = [x for x in subjects if x[1] > 0]
-        sqs_contributor = sqs.facet('contributor')
-        contributors = sqs_contributor.facet_counts()['fields']['contributor'][:self.filterlimit]
-        contributors = [x for x in contributors if x[1] > 0]
-        sqs_type = sqs.facet('resource_type')
-        types = sqs_type.facet_counts()['fields']['resource_type'][:self.filterlimit]
-        types = [x for x in types if x[1] > 0]
-        sqs_availability = sqs.facet('availability')
-        availability = sqs_availability.facet_counts()['fields']['availability'][:self.filterlimit]
-        availability = [x for x in availability if x[1] > 0]
-
-        filterdata = [authors, owners, subjects, contributors, types, availability]
-
         resources = []
 
         # TODO future release will add title and facilitate order_by title_exact
         # convert sqs to list after facet operations to allow for Python sorting instead of Haystack order_by
-        sqs = list(sqs)
         if sort == 'title':
+            sqs = list(sqs)
```

Comment:
why is sqs put into a list for this sort but not the other?

Reply:
sqs is a query object until something like list() is called. To act on the collection, I'm calling list, forcing the eval

Reply:
I know the difference between a query object and a list.  I'm asking why you're making it a list here but not for the reverse sort below.

Reply:
Interesting. Based on testing, looks like the lambda or sorted are handling it and the list() command is superfluous. Originally, it was outside of the if... then... block and was moved inside as a differential experiment to see if it had an impact on performance.